### PR TITLE
Sending the full container path to the UI

### PR DESCRIPTION
**Pull request by freddyucv**
The Multi tree was saved with the default host container id and not with the container's id in the new site, so now it is send the container full path  to the UI, then it is send back by the UI when a Multi tree is saved

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                .host(host)
+                .nextPersisted();
+
+        final FileAssetContainer container = (FileAssetContainer) APILocator.getContainerAPI()
+                .find(fileAssetContainer.getInode(), APILocator.systemUser(), true);
+
+        final ContainerLoader containerLoader = new ContainerLoader();
+
+        final VelocityResourceKey velocityResourceKey = new VelocityResourceKey(fileAssetContainer, PageMode.EDIT_MODE);
+        final InputStream inputStream = containerLoader.writeObject(velocityResourceKey);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java]: Codacy Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java [hunk 2]
```diff
@@ -1,14 +1,49 @@
...
 
+import com.dotcms.datagen.ContainerAsFileDataGen;
+import com.dotcms.datagen.FolderDataGen;
 import com.dotcms.datagen.SiteDataGen;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.portlets.ContentletBaseTest;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.containers.model.FileAssetContainer;
+import com.dotmarketing.portlets.folders.model.Folder;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.folders.model.Folder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java [hunk 3]
```diff
@@ -1,14 +1,49 @@
...
+import com.dotmarketing.util.Constants;
+import com.dotmarketing.util.HostUtil;
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.util.Optional;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Optional'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java [hunk 4]
```diff
@@ -1,14 +1,49 @@
...
+import com.dotcms.datagen.ContainerAsFileDataGen;
+import com.dotcms.datagen.FolderDataGen;
 import com.dotcms.datagen.SiteDataGen;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.portlets.ContentletBaseTest;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.containers.model.FileAssetContainer;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Constants;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.util.Constants'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java [hunk 5]
```diff
@@ -1,14 +1,49 @@
...
+import com.dotmarketing.util.HostUtil;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.util.HostUtil'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java [hunk 6]
```diff
@@ -1,14 +1,49 @@
 package com.dotmarketing.portlets.containers.business;
 
+import com.dotcms.datagen.ContainerAsFileDataGen;
+import com.dotcms.datagen.FolderDataGen;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.FolderDataGen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java [hunk 7]
```diff
@@ -1,14 +1,49 @@
...
 import com.dotcms.datagen.SiteDataGen;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.portlets.ContentletBaseTest;
+import com.dotmarketing.portlets.containers.model.Container;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.containers.model.Container'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ContainerLoaderTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+import com.dotcms.datagen.ContainerAsFileDataGen;
+import com.dotcms.datagen.SiteDataGen;
+import com.dotcms.repackage.org.apache.commons.io.IOUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.containers.model.Container;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.portlets.containers.model.Container'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+import com.dotmarketing.portlets.containers.model.FileAssetContainer;
+import com.dotmarketing.util.PageMode;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.Assert.assertEquals;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4945041)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to test: {@link ContainerLoader#writeObject(VelocityResourceKey)}
+     * when: the File Container's html is build
+     * Should: return a div with the File Comntainer's id
+     */
+    @Test
+    public void aaa() throws DotDataException, DotSecurityException, IOException {
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java]: please rename this test

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+                .host(host)
+                .nextPersisted();
+
+        final FileAssetContainer container = (FileAssetContainer) APILocator.getContainerAPI()
+                .find(fileAssetContainer.getInode(), APILocator.systemUser(), true);
+
+        final ContainerLoader containerLoader = new ContainerLoader();
+
+        final VelocityResourceKey velocityResourceKey = new VelocityResourceKey(fileAssetContainer, PageMode.EDIT_MODE);
+        final InputStream inputStream = containerLoader.writeObject(velocityResourceKey);
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java]: this InputStream should be closed in a finally statement or in a try with resources

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java (hunk 11 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java]: done https://github.com/dotCMS/core/pull/17950/commits/e8ef923ad798cfa1c6428e2516f90183bcd40a4e#diff-90a1b8ea255c50a142e8987ff9b84049R37

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java (hunk 12 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/ContainerLoaderTest.java]: done https://github.com/dotCMS/core/pull/17950/commits/e8ef923ad798cfa1c6428e2516f90183bcd40a4e#diff-90a1b8ea255c50a142e8987ff9b84049R52

**Review by nollymar — APPROVED** [review]

**Pull request merged**